File: backend/src/classflow/domain/entities/feedback.py
from dataclasses import dataclass, field
from datetime import datetime

from classflow.domain.exceptions import ValidationError


@dataclass
class Feedback:
    id: int = field(init=False)
    organization_id: int = field(init=False)
    author_id: int
    rating: int
    teacher_id: int | None = None
    course_id: int | None = None
    text: str = ''
    is_active: bool = True
    created_at: datetime = field(init=False)

    def __post_init__(self) -> None:
        if self.rating < 1 or self.rating > 5:
            raise ValidationError('rating must be between 1 and 5')

        if not self.teacher_id and not self.course_id:
            raise ValidationError(
                'teacher_id ot course_id id must be provided',
            )

        if self.teacher_id and self.course_id:
            raise ValidationError(
                'teacher_id and course_id are mutually exclusive',
            )

File: frontend/src/entities/feedback.ts
import type { OrganizationMemberDetail } from './organization.ts';
import type { CourseDetail } from './course.ts';

interface BaseFeedback {
  id: string;
  rating: number;
  text: string;
  createdAt: string;
}

export interface Feedback extends BaseFeedback {
  authorId: number;
  teacherId?: number;
  courseId?: number;
}

export interface FeedbackDetail extends Feedback {
  author: OrganizationMemberDetail;
  teacher: OrganizationMemberDetail;
  course: CourseDetail;
}
